src/firefly/js/charts/ui/ColSelectView.jsx

```diff
@@ -154,6 +156,30 @@ function renderCloseAndHelpButtons(tblId,onColSelected,buttonText,popupId) {
 );
 }
 
+function setSelectedColumns(tblId,uploadInfo,onColSelected) {
+    const tbl = getTableUiByTblId(tblId);
+    const data = tbl.data; //columns
+    const selectInfo = tbl.selectInfo;
+    const selectedIndices = [];
+
+    for (const [idx, entry] of data.entries()) {
+        if (selectInfo.selectAll) { //ignore selectInfo.exceptions
+            //go through data - for each idx NOT in selectInfo.exceptions, grab data[currIdx][5] - 5 is the idx in uploadInfo.columns
+            if (!(selectInfo.exceptions).has(idx)) selectedIndices.push(data[idx][5]);
+        }
+        else { //selectInfo.exceptions are the indices we want to select
+            if ((selectInfo.exceptions).has(idx)) selectedIndices.push(data[idx][5]);
+        }
+    }
```

Comment:
I may need to change this part. There is probably a better way of getting the selected columns info (I did it this way for now to get around the user filtering the columns and then making their selections).

Reply:
Yes, there is.  `firefly.util.table.getSelectedData`